tests/test_proxy_functional.py

```diff
@@ -1,14 +1,137 @@
 import asyncio
 import os
 import pathlib
+import ssl
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
+# @pytest.fixture
+# def secure_proxy_url(
+#     aiohttp_unused_port, tls_ca_certificate_pem_path, tls_ca_private_key_pem_path
+# ):
+#     """Return the an instance of a running secure proxy."""
+#     unused_port = str(aiohttp_unused_port())
+#     address = "127.0.0.1"
+#     input_args = [
+#         "--num-workers",
+#         "1",
+#         "--hostname",
+#         "127.0.0.1",
+#         "--port",
+#         unused_port,
+#         "--cert-file",
+#         tls_ca_certificate_pem_path,
+#         "--key-file",
+#         tls_ca_private_key_pem_path,
+#     ]
+#     proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+#     with proxy.Proxy(input_args=input_args):
+#         yield proxy_url
+
+
+@pytest.fixture
+def secure_proxy_url(
+    aiohttp_unused_port,
+    tls_certificate,
+):
+    """Return the an instance of a running secure proxy."""
+    unused_port = str(aiohttp_unused_port())
+    address = "127.0.0.1"
+    with tls_certificate.cert_chain_pems[0].tempfile() as server_cert_path:
+        with tls_certificate.private_key_pem.tempfile() as server_key:
+            input_args = [
+                "--num-workers",
+                "1",
+                "--hostname",
+                "127.0.0.1",
+                "--port",
+                unused_port,
+                "--cert-file",
+                server_cert_path,
+                "--key-file",
+                server_key,
+            ]
+            proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+            with proxy.Proxy(input_args=input_args):
+                yield proxy_url
+
+
+# @pytest.mark.xfail
+async def test_secure_proxy_http_absolute_path(
```

Comment:
This test is fully working and is more-or-less in its final form. It just needs the xfail uncomented from above, and possibly paramterize'd on the protocol to be `http` versus `https` if the tests can be consolidated.